### PR TITLE
server: record version in node

### PR DESCRIPTION
**Pull request by nolouch**
## What have you changed? (mandatory)
for report the version of this node. record the version.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)
 manual tests

## Refer to a related PR or issue link (optional)
https://github.com/pingcap/pd/pull/1148

### CODE REVIEW

File: src/server/node.rs [hunk 1]
```diff
@@ -105,6 +105,7 @@ where
         } else {
             store.set_address(cfg.advertise_addr.clone())
         }
+        store.set_version(env!("CARGO_PKG_VERSION").to_string());
```
Comment (zhangjinpeng1987) [inline src/server/node.rs]: Can you explain how versions works? Why we should use `CARGO_PKG_VERSION`?

**Comment by siddontang** [comment]
I still prefer number rather than string, but seems that getting version with string is more easy and I don't want to introduce a lib to parse the SemVer in TiKV, so LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay @disksing

File: src/server/node.rs (hunk 1 above)
Comment (siddontang) [reply src/server/node.rs]: @zhangjinpeng1987 PD needs it to handle the compatibility when rolling update.

Comment (nolouch) [reply src/server/node.rs]: PD will maintain a cluster version. If rolling upgrade finished, the cluster version will increase. the Incompatible tikv with the cluster version will not not register successfully at startup.

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by overvenus** [comment]
@siddontang Actually we dont need an extra crate to parse semver, cargo already gives us:
 - `CARGO_PKG_VERSION_MAJOR`
 - `CARGO_PKG_VERSION_MINOR`
 - `CARGO_PKG_VERSION_PATCH`

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

**Comment by siddontang** [comment]
great @overvenus

**Pull request merged**